js/modules/k6/marshalling_test.go

```diff
@@ -0,0 +1,146 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package k6_test
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/loadimpact/k6/js"
+	"github.com/loadimpact/k6/lib"
+	"github.com/loadimpact/k6/lib/testutils"
+	"github.com/loadimpact/k6/lib/types"
+	"github.com/loadimpact/k6/stats"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	null "gopkg.in/guregu/null.v3"
+)
+
+func TestSetupDataMarshalling(t *testing.T) {
+	tb := testutils.NewHTTPMultiBin(t)
+	defer tb.Cleanup()
+
+	script := []byte(tb.Replacer.Replace(`
+		import http from "k6/http";
+		import html from "k6/html";
+		import ws from "k6/ws";
+		export let options = { setupTimeout: "10s" };
+
+
+		function make_jar() {
+			let jar = http.cookieJar();
+			jar.set("test", "something")
+			return jar;
+		}
+
+		export function setup() {
+			let res = http.get("http://test.loadimpact.com:HTTPBIN_PORT");
```

Comment:
Why use `test.loadimpact.com` here and below? Wouldn't [`HTTPBIN_URL`](https://github.com/loadimpact/k6/blob/bb6e97542c63b1282126c650f6b757c5ecf9d2b1/lib/testutils/httpmultibin.go#L155-L164) also work, so you don't override the `Hosts` below? Or even maybe `HTTPSBIN_URL` to trigger the OCSP check?